lib/org/openpsa/sales/handler/action.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * @package org.openpsa.sales
+ * @author The Midgard Project, http://www.midgard-project.org
+ * @copyright The Midgard Project, http://www.midgard-project.org
+ * @license http://www.gnu.org/licenses/lgpl.html GNU Lesser General Public License
+ */
+
+/**
+ * Sales action handler ***creates pdf-file***
+ *
+ * @package org.openpsa.sales
+ */
+class org_openpsa_sales_handler_sales_action extends midcom_baseclasses_components_handler
+{
+    private $salesproject;
+
+    public function _on_initialize()
+    {
+        midcom::get()->auth->require_valid_user();
+        if (empty($_POST['id'])) {
+            throw new midcom_error('Incomplete POST data');
+        }
+        $this->salesproject = new org_openpsa_sales_salesproject_dba((int) $_POST['id']);
+        $this->salesproject->require_do('midgard:update');
+    }
+
+    private function reply($success, $message)
+    {
+        $message = array(
+            'title' => $this->_l10n->get($this->_component),
+            'type' => $success ? 'info' : 'error',
+            'message' => $message
+        );
+
+        if (!empty($_POST['relocate'])) {
+            midcom::get()->uimessages->add($message['title'], $message['message'], $message['type']);
+            return new midcom_response_relocate('salesproject/' . $this->salesproject->guid . '/');
+        }
+
+        $result = array(
+            'success' => $success,
+            'action' => $this->_master->render_invoice_actions($this->salesproject),
+            'due' => strftime('%Y-%m-%d', $this->salesproject->due),
+            'new_status' => $this->salesproject->get_status(),
+            'message' => $message
+        );
+        return new midcom_response_json($result);
```

Comment:
Is this code even reachable? I didn't see anything that would do AJAX requests in this PR